tests/workflows/update/update.go

```diff
@@ -0,0 +1,44 @@
+package update
+
+import (
+	"fmt"
+
+	"go.temporal.io/sdk/workflow"
+)
+
+const (
+	FetchAndAdd = "fetch_and_add"
+	Done        = "done"
+)
+
+func Counter(ctx workflow.Context, val int) (int, error) {
+	log := workflow.GetLogger(ctx)
+	counter := val
+
+	if err := workflow.SetUpdateHandlerWithOptions(
+		ctx,
+		FetchAndAdd,
+		func(ctx workflow.Context, i int) (int, error) {
+			tmp := counter
+			counter += i
+			log.Info("counter updated", "added", i, "new-value", counter)
```

Comment:
It's a small thing, but this is a new workflow.Context, so you should probably get the logger here